client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetMappingsResponseTests.java

```diff
@@ -19,54 +19,40 @@
-import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
 import org.elasticsearch.index.mapper.MapperService;
-import org.elasticsearch.rest.BaseRestHandler;
-import org.elasticsearch.test.ESTestCase;
 
 import java.io.IOException;
-import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
-import java.util.function.Predicate;
 
-import static org.elasticsearch.client.indices.GetMappingsResponse.MAPPINGS;
-import static org.elasticsearch.test.AbstractXContentTestCase.xContentTester;
+public class GetMappingsResponseTests
+    extends AbstractResponseTestCase<org.elasticsearch.action.admin.indices.mapping.get.GetMappingsResponse, GetMappingsResponse> {
 
-public class GetMappingsResponseTests extends ESTestCase {
-
-    // Because the client-side class does not have a toXContent method, we test xContent serialization by creating
-    // a random client object, converting it to a server object then serializing it to xContent, and finally
-    // parsing it back as a client object. We check equality between the original client object, and the parsed one.
-    public void testFromXContent() throws IOException {
-        xContentTester(
-            this::createParser,
-            GetMappingsResponseTests::createTestInstance,
-            GetMappingsResponseTests::toXContent,
-            GetMappingsResponse::fromXContent)
-            .supportsUnknownFields(true)
-            .assertEqualsConsumer(GetMappingsResponseTests::assertEqualInstances)
-            .randomFieldsExcludeFilter(randomFieldsExcludeFilter())
-            .test();
-    }
-
-    private static GetMappingsResponse createTestInstance() {
-        Map<String, MappingMetaData> mappings = Collections.singletonMap(
-            "index-" + randomAlphaOfLength(5), randomMappingMetaData());
-        return new GetMappingsResponse(mappings);
+    @Override
+    protected org.elasticsearch.action.admin.indices.mapping.get.GetMappingsResponse createServerTestInstance(XContentType xContentType) {
+        ImmutableOpenMap.Builder<String, MappingMetaData> mappings = ImmutableOpenMap.builder();
+        int numberOfIndexes = randomIntBetween(1, 5);
+        for (int i = 0; i < numberOfIndexes; i++) {
+            mappings.put("index-" + randomAlphaOfLength(5), randomMappingMetaData());
+        }
+        return new org.elasticsearch.action.admin.indices.mapping.get.GetMappingsResponse(mappings.build());
     }
 
-    private static void assertEqualInstances(GetMappingsResponse expected, GetMappingsResponse actual) {
-        assertEquals(expected.mappings(), actual.mappings());
+    @Override
+    protected GetMappingsResponse doParseToClientInstance(XContentParser parser) throws IOException {
+        return GetMappingsResponse.fromXContent(parser);
     }
 
-    private Predicate<String> randomFieldsExcludeFilter() {
```

Comment:
It could make sense to keep this as a precaution, to ensure the mappings are always well-formed.

Reply:
HLRC's AbstractResponseTestCase doesn't seem to support this, unfortunately

Reply:
👍